### PR TITLE
fix #138963 rocketnews24.com,youpouch.com,soranews24.com

### LINKED ISSUE #138963: rocketnews24.com
### Issue URL (Incorrect Blocking)
 [https://rocketnews24.com/2023/01/02/1743041/](https://adguardteam.github.io/AnonymousRedirect/redirect.html?url=https%3A%2F%2Frocketnews24.com%2F2023%2F01%2F02%2F1743041%2F)

### Screenshots
<details>
  <summary>Screenshot 1</summary>

  Screenshot 1
</details>
<details>
  <summary>Screenshot 2</summary>

  Screenshot 2
</details>

### System configuration

Information | value
--- | ---
AdGuard product: | AdGuard/AdGuard Pro for iOS v4.4.5
Browser: | Safari
DNS filtering: | server: `tls://dns.adguard-dns.com`<br>filters: `https://adguardteam.github.io/AdGuardSDNSFilter/Filters/filter.txt`
Filters: | <b>Ad Blocking:</b><br/>AdGuard Base, <br/>AdGuard Mobile Ads, <br/>EasyList<br/><br/><b>Privacy:</b><br/>AdGuard Tracking Protection<br/><br/><b>Social Widgets:</b><br/>AdGuard Social Media<br/><br/><b>Annoyances:</b><br/>AdGuard Annoyances<br/><br/><b>Security:</b><br/>Online Malicious URL Blocklist<br/><br/><b>Other:</b><br/>AdGuard Experimental, <br/>AdGuard DNS<br/><br/><b>Language-specific:</b><br/>AdGuard Russian, <br/>AdGuard Japanese, <br/>AdGuard Spanish/Portuguese, <br/>AdGuard French, <br/>AdGuard Chinese, <br/>List-KR

### PR DESCRIPTION
**Pull request by Yuki2718**
# Creating the pull request

> Please include a summary of the change and which issue is fixed\
> If the related issue does not exist in our repository, please create it before making pull request\
> It is highly recommended to use our [Web Reporting Tool](https://kb.adguard.com/en/technical-support/reporting-tool) instead of creating an issue on GitHub directly\
> Please note, that we verify every pull request manually, so it may take time to apply it

### To avoid invalid pull requests, please check and confirm following terms

- [x] This is not an ad/bug report;
- [x] My code follows the [guidelines](https://github.com/AdguardTeam/AdguardFilters/blob/master/CONTRIBUTING.md) and [syntax](https://kb.adguard.com/general/how-to-create-your-own-ad-filters) of this project;
- [x] I have performed a self-review of my own changes;
- [x] My changes do not break web sites, apps and files structure.

### If the problem does not fall under any category that is listed here, please write a comment below in corresponding section

- [ ] Missed ads or ad leftovers;
- [ ] Website or app doesn't work properly;
- [x] AdGuard gets detected on a website;
- [ ] Missed analytics or tracker;
- [ ] Social media buttons — share, like, tweet, etc;
- [ ] Annoyances — pop-ups, cookie warnings, etc;
- [ ] Filters maintenance.

### Enter the issue address

https://github.com/AdguardTeam/AdguardFilters/issues/138963

### Add your comment and screenshots

1. Your comment

New idea, prevent them to override images. Please test on iOS with DNS filtering enabled. Removed `replace=/\w+\.onload=//` as no more needed.

### Terms

- [x] By submitting this issue, I agree that pull request does not contain private info and all conditions are met

### CODE REVIEW

File: JapaneseFilter/sections/antiadblock.txt [hunk 1]
```diff
@@ -47,20 +47,12 @@ rocketnews24.com,youpouch.com#%#//scriptlet('remove-attr', 'id', '#div-gpt-ad-re
 @@||rocketnews24.com^$generichide
 ||googlesyndication.com/safeframe/$subdocument,redirect=noopframe,domain=rocketnews24.com|youpouch.com
 ||fundingchoicesmessages.google.com^$script,redirect-rule=noopjs,domain=rocketnews24.com|youpouch.com
-!#if (adguard_app_windows || adguard_app_mac || adguard_app_android || adguard_ext_firefox)
-||rocketnews24.com/_static/$script,replace=/\w+\.onload=//
-||youpouch.com/_static/$script,replace=/\w+\.onload=//
-!#endif
-!#if (adguard_app_windows || adguard_app_mac || adguard_app_android)
+!#if (adguard_app_windows || adguard_app_mac || adguard_app_android || adguard_app_ios || adguard_ext_safari || adguard_ext_chromium || adguard_ext_edge)
```
Comment (Alex-302) [inline JapaneseFilter/sections/antiadblock.txt:50]: Directive is unnecessary here.

Comment (Yuki2718) [reply JapaneseFilter/sections/antiadblock.txt:50]: Why?
> the scriptlet rule alone breaks images.

Comment (Yuki2718) [reply JapaneseFilter/sections/antiadblock.txt:50]: The prevent-setTimeout rule should be applied if and only if `#%#AG_onLoad(function(){let a=document.querySelectorAll("img+noscript");let ns=function(b){let f=new DOMParser();let g=f.parseFromString(b.textContent,'text/html');return document.adoptNode(g.querySelector('img'));};for(let b of a){let c=ns(b);b.parentNode.replaceChild(c,b);}});` is applied.

Comment (krystian3w) [reply JapaneseFilter/sections/antiadblock.txt:50]: Maybe `if` blocks compile new Addon for Opera and Firefox (bumped version without new pure js rule).

Comment (Alex-302) [reply JapaneseFilter/sections/antiadblock.txt:50]: `$replace` works in apps only (FF has partial support)
`#%#` works with all of these plathorms, + Opera + FF. Maybe can be checked to this?
`!#if (adguard && !adguard_ext_firefox && !adguard_ext_opera)`

Comment (Yuki2718) [reply JapaneseFilter/sections/antiadblock.txt:50]: ext_ublock too. Will change.

**Review by Alex-302 — APPROVED** [review]

**Pull request merged**